wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -127,19 +138,16 @@ final class DeviceInfoViewModel: ObservableObject {
 
     @MainActor
     func removeDevice() async {
-        let isRemoved = await actionsHandler.removeDevice()
-        self.isRemoved = isRemoved
+        self.shouldDismiss = await actionsHandler.removeDevice()
     }
 
     func resetSession() {
         actionsHandler.resetSession()
     }
 
+    @MainActor
     func updateVerifiedStatus(_ value: Bool) async {
```

Comment:
Should we add `@MainActor` attribute to the whole view model (e.g. UIViewController has this too)? In my experience this is easier and less error-prune than adding it to single funcs here and there, but both has advantages.

Reply:
Generally refrain from doing that as some of the functions could be fire and forget. Not an optimisation we should do IMHO